src/app/product-detail/product-detail.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { ProductService } from '../api/products.service';
+import { IProduct } from '../Interface/products';
+
+@Component({
+  selector: 'app-product-detail',
+  templateUrl: './product-detail.component.html',
+  styleUrls: ['./product-detail.component.scss']
+})
+export class ProductDetailComponent implements OnInit, OnDestroy {
+
+  productId: number = 0;
+
+  products: IProduct[] = [];
+
+  sub!: Subscription;
+
+  isDataProvided: boolean = true;
+
+  selectedProduct: IProduct | undefined;
+
+  constructor(private route: ActivatedRoute,
+                  public productService: ProductService) {}
+
+  ngOnInit(): void {
+    const id = Number(this.route.snapshot.paramMap.get("id"));
+    this.productId = id;
+    console.log('Product ID', this.productId)
+
+    this.sub = this.productService.productData$.subscribe(
+      data => { 
+          if(data.length === 0) {
+              this.isDataProvided = false;
+          } else {
+              this.products = data;
+              this.selectedProduct = this.products.find(el => el.id == id);
```

Comment:
Please always use === and make sure the types match

Reply:
ok